File: src/services/adminService.js
import api from './api';

export const adminService = {
    // Dashboard
    getDashboardStats: () => api.get('/admin/dashboard'),

    // Users
    getAllUsers: () => api.get('/admin/users'),
    blockUser: (userId, active) => api.post(`/admin/users/${userId}/status?active=${active}`),

    // Recruiters
    getAllRecruiters: () => api.get('/admin/recruiters'),
    approveRecruiter: (id) => api.post(`/admin/recruiters/${id}/approve`),
    blockRecruiter: (userId, active) => api.post(`/admin/users/${userId}/status?active=${active}`),

    // Jobs
    getAllJobs: () => api.get('/admin/jobs'),
    getPendingJobs: () => api.get('/admin/jobs/pending'),
    approveJob: (id) => api.post(`/admin/jobs/${id}/approve`),
    rejectJob: (id) => api.post(`/admin/jobs/${id}/reject`),

    // Applications
    getAllApplications: () => api.get('/admin/applications'),
};

File: backend/src/main/java/com/luzo_portal/service/AdminService.java
package com.luzo_portal.service;

import com.luzo_portal.dto.AdminDashboardStats;
import com.luzo_portal.dto.JobResponse;
import com.luzo_portal.dto.RecruiterProfileResponse;
import com.luzo_portal.dto.UserResponse;

import java.util.List;

public interface AdminService {

    AdminDashboardStats getDashboardStats();

    List<JobResponse> getPendingJobs();

    JobResponse approveJob(Long jobId);

    void deleteJob(Long jobId);

    void approveRecruiter(Long recruiterProfileId);

    void blockUser(Long userId, boolean active);

    List<UserResponse> getAllUsers();

    List<RecruiterProfileResponse> getAllRecruiters();

    List<JobResponse> getAllJobs();

    void resolveReport(Long reportId, String adminNotes);
}